src/OpenLoco/src/Map/MapGenerator.cpp

```diff
@@ -392,6 +400,112 @@ namespace OpenLoco::World::MapGenerator
+        {
+            auto basePath = Platform::getUserDirectory();
+
+            // open ui prompt window
+            // doesn't work...
+            bool confirm = Ui::Windows::PromptBrowse::open(OpenLoco::Ui::Windows::PromptBrowse::browse_type::load, (char*)basePath.string().c_str(), "*.png", StringIds::title_prompt_load_game);
+
+            return confirm;
+        }
+
+        // note: png image must be 384x384 or otherwise this method will probably crash
+        int generateFromHeightmapPng(HeightMapRange heightMap)
+        {
+            auto basePath = Platform::getUserDirectory();
+            auto defaultHeightmap = basePath / "heightmap.png";
+            auto filename = defaultHeightmap.string();
+
+            // todo:
+            /*if (openUiPngBrowser())
+            {
+                filename = std::string(_savePath);
+            }*/
+
+            std::ifstream inFile(filename, std::ios::binary);
+
+            png_structp png = png_create_read_struct(PNG_LIBPNG_VER_STRING, NULL, NULL, NULL);
+            if (!png)
+            {
+                std::cerr << "Failed to create PNG read struct" << std::endl;
+                return 1;
+            }
+
+            png_infop info = png_create_info_struct(png);
+            if (!info)
+            {
+                std::cerr << "Failed to create PNG info struct" << std::endl;
+                png_destroy_read_struct(&png, NULL, NULL);
+                return 1;
+            }
+
+            // Set error handler
+            if (setjmp(png_jmpbuf(png)))
+            {
+                png_destroy_read_struct(&png, &info, NULL);
+                throw std::runtime_error("PNG ERROR");
+            }
+
+            FILE* file = fopen(filename.c_str(), "rb");
+            png_init_io(png, file);
+            png_read_info(png, info);
+
+            int width = png_get_image_width(png, info);
+            int height = png_get_image_height(png, info);
+            // png_byte colorType = png_get_color_type(png, info);
+
+            std::vector<png_byte> imageData(width * height * 4);
+
+            png_bytep* rowPointers = new png_bytep[height];
+            for (int y = 0; y < height; y++)
+            {
+                rowPointers[y] = &imageData[y * width * 4];
+            }
+            png_read_image(png, rowPointers);
+
+            // do stuff with the image
+            png_byte red, green, blue, alpha;
+
+            for (int32_t y = 0; y < height; y++)
+            {
+                for (int32_t x = 0; x < width; x++)
+                {
+                    getPixel(imageData, x, y, red, green, blue, alpha, width);
+                    // flipped x,y is intentional here; is openloco flipping them somewhere else down the line coz otherwise this image is flipped
+                    heightMap[{ y, x }] = ((red + green + blue) / 3) / 2; // scale down to something openloco is happy with - todo: figure out openloco's scale. 180m seems max land height
```

Comment:
can probably just use one channel here since we're expecting a grayscale image, but this is just how i did it here